FormHoneypotPlugin.inc.php

```diff
@@ -278,15 +326,26 @@ function getTemplatePath($inCore = false) {
 	 */
 	function handleTemplateDisplay($hookName, $args) {
 		
-		$templateMgr = $args[0];
+		// Could get this from $args[0] in this case but the result is identical
```

Comment:
How can we be sure of this?  Couldn't the caller have made changes to the passed `TemplateManager` object which we, or another hook, might rely on?  What if some other plugin has registered an output filter on the `TemplateManager` in context and now we are instantiating a new `TemplateManager` where we are registering our own filter?

This is especially critical because we are not passing a `PKPRequest` object to the `TemplateManager::getManager()` call.  See:
https://github.com/pkp/pkp-lib/blob/13fd3e983a76fb50ef4d134ddf0ba9b5cf404c7e/classes/template/PKPTemplateManager.inc.php#L943-L967

I would leave the original logic of grabbing the `TemplateManager` from the hook arguments.  True, the arguments are subject to change, but there is hand-waiving voodoo under the static call.

Reply:
That's fair. It's a decision made a while ago, with a worse understanding of what was going on behind the scenes. Reverted in 31f0fce.